### PR TITLE
Remove proto/api submodule

### PR DESCRIPTION
**Pull request by dnr**
## What changed?
- Remove `proto/api` submodule.
- Adds a new program that's automatically run by the Makefile as required to output the api protos in binary form, which can be used by most tools. The command is run by `go run` and uses the same api-go reference as the rest of the server for the source protos.
- Removed http annotations from internal matchingservice proto.

## Why?
- Ensure that the api-go import (in go.mod) and the proto files imported by this repo's proto files are always in sync.
- Simplify update process, stop having to mess with submodules.

## How did you test it?
running commands manually a lot

## Potential risks
Importing protos from somewhere other than our api and google well-known protos may require some updates to the script's import path logic.

## Documentation
updated docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -233,22 +229,15 @@ $(PROTO_OUT):
...
-update-proto-submodule:
-	@printf $(COLOR) "Update proto submodule from remote..."
-	git submodule update --force --remote $(PROTO_ROOT)/api
-
-install-proto-submodule:
-	@printf $(COLOR) "Install proto submodule..."
-	git submodule update --init $(PROTO_ROOT)/api
-
 protoc: clean-proto $(PROTO_OUT) $(PROTOGEN) $(PROTOC_GEN_GO) $(PROTOC_GEN_GO_GRPC) $(PROTOC_GEN_GO_HELPERS)
+	@./cmd/tools/getproto/run.sh --out $(API_BINPB)
```
Comment (tdeebswihart) [inline Makefile:233]: $(API_BINPB) should be a dependency of the protoc stage

File: Makefile [hunk 2]
```diff
@@ -233,22 +229,15 @@ $(PROTO_OUT):
...
 	@$(PROTOGEN) \
-		-I=proto/api \
-		-I=proto/dependencies \
+		--descriptor_set_in=$(API_BINPB) \
 		--root=proto/internal \
 		--rewrite-enum=BuildId_State:BuildId \
 		-p go-grpc_out=$(PROTO_PATHS) \
 		-p go-helpers_out=$(PROTO_PATHS)
+	@rm -f $(API_BINPB)
```
Comment (tdeebswihart) [inline Makefile:240]: Why are we deleting API_BINPB after using it?

File: Makefile [hunk 3]
```diff
@@ -337,19 +326,24 @@ lint: lint-code lint-actions lint-api lint-protos
 
 lint-api: $(API_LINTER)
 	@printf $(COLOR) "Linting proto API..."
-	$(call silent_exec, $(API_LINTER) --set-exit-status $(PROTO_IMPORTS) --config=$(PROTO_ROOT)/api-linter.yaml $(PROTO_FILES))
+	@./cmd/tools/getproto/run.sh --out $(API_BINPB)
```
Comment (tdeebswihart) [inline Makefile:329]: I don't like how we're regenerating this and deleting it constantly. This should be tracked as a target dependency that we regenerate as needed (say, when the `api-go` version in the go.mod changes)

Comment (dnr) [reply Makefile:329]: Ok, that's fair. The argument for not putting it in the repo is that it's a 120K binary file that changes from time to time, and is totally (+quickly) derivable from stuff in the repo. But it probably makes sense.

Comment (tdeebswihart) [reply Makefile:329]: We can derive it and put it in an ignored subfolder like we do the binary dependencies we install :)

Comment (dnr) [reply Makefile:329]: Oh I thought you meant include it in the repo. Well I already put it in .gitignore, so we could just leave it there.

File: Makefile [hunk 4]
```diff
@@ -335,21 +325,22 @@ lint-code: $(GOLANGCI_LINT)
...
+	@protoc --descriptor_set_in=$(API_BINPB) -I=proto/internal $(PROTO_FILES) -o /dev/stdout | (cd proto/internal && $(ROOT)/$(BUF) lint -)
+
+# TODO: fix this to work with getproto + API_BINPB
+# buf-build: $(BUF)
+# 	@printf $(COLOR) "Build image.bin with buf..."
+# 	@(cd $(PROTO_ROOT) && $(ROOT)/$(BUF) build -o image.bin)
+#
+# buf-breaking: $(BUF)
+# 	@printf $(COLOR) "Run buf breaking changes check against image.bin..."
+# 	@(cd $(PROTO_ROOT) && $(ROOT)/$(BUF) breaking --against image.bin)
```
Comment (tdeebswihart) [inline Makefile:343]: Do we rely on these for CI? I'd at least like the breaking changes check to work if at all possible

Comment (dnr) [reply Makefile:343]: They're unused as far as I can tell. image.bin is ignored in .gitignore which makes it kind of useless. If we did want to do this (which we do)... we should probably use buf's git support instead to check across branches. An in-repo file only allows linear checks so doesn't prevent cross-branch problems.

**Review by tdeebswihart — COMMENTED** [review]
Your protogen version is out of date as we pin protogen in the makefile; it's separate from the Go API version in go.mod

**Comment by dnr** [comment]
> Your protogen version is out of date as we pin protogen in the makefile; it's separate from the Go API version in go.mod

I know, I'm waiting to do an api release so I can bump it

**Review by tdeebswihart — APPROVED** [review]

**Pull request merged**